src/components/Envoy/writers/BaseWriter.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { ICell, ISortBy, SortByDirection, Table, TableBody, TableHeader } from '@patternfly/react-table';
+import { ClusterSummaryWriter, ClusterWriter } from './ClusterWriter';
+import { RouteSummaryWriter, RouteWriter } from './RouteWriter';
+import { ListenerSummaryWriter, ListenerWriter } from './ListenerWriter';
+
+export interface SummaryWriter {
+  head: () => ICell[];
+  rows: () => (string | number)[][];
+  setSorting: (columnIndex: number, direction: string) => void;
+}
+
+export function SummaryWriterRenderer<T extends SummaryWriter>() {
+  interface SummaryWriterProps<T> {
+    writer: T;
+  }
+
+  interface SummaryWriterState {
+    sortBy: ISortBy;
+  }
+
+  return class SummaryWriter extends React.Component<SummaryWriterProps<T>, SummaryWriterState> {
+    constructor(props) {
+      super(props);
+      this.state = {
+        sortBy: {
+          direction: SortByDirection.asc,
+          index: 0
+        }
+      };
+    }
+
+    onSort = (_: React.MouseEvent, columnIndex: number, sortByDirection: SortByDirection) => {
+      this.props.writer.setSorting(columnIndex, sortByDirection);
+      this.setState({
+        sortBy: {
+          direction: sortByDirection,
+          index: columnIndex
+        }
+      });
+    };
+
+    render() {
+      return (
+        <Table
+          aria-label="Sortable Table"
+          cells={this.props.writer.head()}
+          rows={this.props.writer.rows()}
+          sortBy={this.state.sortBy}
+          onSort={this.onSort}
+        >
+          <TableHeader />
+          <TableBody />
+        </Table>
+      );
+    }
+  };
+}
+
+export const SummaryWriterBuilder = (resource: string, config: any) => {
```

Comment:
I know naming is hard and opinionated :).
Coming from Java background, I read "Writer" very confusing as normally it's used for an abstraction of streams data.

If this is just a table and we are in the UI, a suggestion, "SummaryTable", "ListenersTable" or using "*List" like "SummaryList", "ListenerLists".

I think that's the way we are naming these patterns in other components and just to follow it to not introduce a new one.